integration/experiment/power_sweep/gen_policy_recommendation.py

```diff
@@ -0,0 +1,287 @@
+            count[pl] = 1
+
+    rtx = {} ; rtxx = {}
+    enx = {} ; enxx = {}
+    for pl in count:
+        rtx[pl] = 0 ; rtxx[pl] = 0
+        enx[pl] = 0 ; enxx[pl] = 0
+
+
+    for pl, rt, en in sorted(simplified_data):
+        rtx[pl] += rt
+        rtxx[pl] += rt**2
+        enx[pl] += en
+        enxx[pl] += en**2
+
+    for pl in count:
+        rtx[pl] /= count[pl]
+        rtxx[pl] /= count[pl]
+        enx[pl] /= count[pl]
+        enxx[pl] /= count[pl]
+
+    stats_out = open(fname, "w")
+    for pl in sorted(count.keys()):
+        stats_out.write("%d %f %f %f %f\n" % (pl, rtx[pl], (rtxx[pl] - rtx[pl]**2)**.5, enx[pl], (enxx[pl] - enx[pl]**2)**.5))
+    stats_out.close()
+
+
+def analyze_extracted_data(simplified_data, rtmodel, enmodel):
+    rtmodel.train([pl for pl, _, __ in simplified_data],
+            [rt for _, rt, __ in simplified_data])
+    enmodel.train([pl for pl, _, __ in simplified_data],
+            [en for _, __, en in simplified_data])
+
+
+def policy_min_energy(plrange, enmodel, rtmodel=None, pltdp=None, max_degradation=None):
+    if pltdp is None:
+        pltdp = max(plrange)
+
+    en_predictions = enmodel.batch_test(plrange)
+    if max_degradation is None:
+        # we don't need the runtime model in this case
+        besten, bestpl = min(zip(en_predictions, plrange))
+        if rtmodel:
+            bestrt = rtmodel.test(bestpl)
+        else:
+            bestrt = None
+        return bestpl, bestrt, besten
+    else:
+        rt_predictions = rtmodel.batch_test(plrange)
+        rt_at_tdp = rtmodel.test(pltdp)
+        constrained_values = [
+                (en, pl, rt)
+                for pl, rt, en in zip(plrange, rt_predictions, en_predictions)
+                if rt <= (1 + max_degradation) * rt_at_tdp]
+        besten, bestpl, bestrt = min(constrained_values)
+        return bestpl, bestrt, besten
+
+
+def main(argv):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--path', help='path containing reports and machine.json', required=True)
+    parser.add_argument('--region_filter', help='comma-separated list of regions to include, default to include all regions', default=None)
+    parser.add_argument('--dump_prefix', help='prefix to dump statistics to, empty to not dump stats')
+    parser.add_argument('--min_pl', help='minimum power limit, default behavior is to use machine.json', default=-1, type=float)
+    parser.add_argument('--max_pl', help='maximum power limit, default behavior is to use machine.json', default=-1, type=float)
+    parser.add_argument('--tdp', help='tdp (sticker power limit), default behavior is to use machine.json', default=-1, type=float)
+    parser.add_argument('--max_degradation', help='maximum allowed runtime degradation, default is 0.1 (i.e., 10%%)', default=0.1, type=float)
+    parser.add_argument('--min_energy', help='ignore max degradation, just give the minimum energy possible', action='store_true')
+
+    args = parser.parse_args(argv[1:])
```

Comment:
Consider refactoring this so this is in the ```if __name__``` section below.  Look at https://github.com/geopm/geopm/blob/dev/integration/experiment/power_sweep/gen_plot_balancer_comparison.py as an example.

There may be some common args you could use here as well.

Reply:
Done.